main/src/ca/uwaterloo/flix/language/phase/unification/set/SetFormula.scala

```diff
@@ -869,90 +754,13 @@ object SetFormula {
-      val elemNeg = elemNeg0.map(instElemSet(_, insts)) match {
-        case Some(e) => e match {
-          case Univ => None
-          case Empty => return Univ
-          case e@ElemSet(_) => Some(e)
-          case other =>
-            // Unreachable since `instElemSet` only returns univ, empty, or elem set.
-            throw InternalCompilerException(s"Unexpected formula: '$other'", SourceLocation.Unknown)
-        }
-        case None => None
-      }
-      if (cstsNeg0.exists(c => insts.get(c.c).contains(Empty))) return Univ
-      if (varsNeg0.exists(x => insts.get(x.x).contains(Empty))) return Univ
-
-      // Compute constants and variables by removing constants and variables that are redundant.
-      // We know from previous checks that none are mapped to the respective short-circuit element.
-      val cstsPos = cstsPos0.filterNot(c => insts.get(c.c).contains(Empty))
-      val varsPos = varsPos0.filterNot(x => insts.get(x.x).contains(Empty))
-      val cstsNeg = cstsNeg0.filterNot(c => insts.get(c.c).contains(Univ))
-      val varsNeg = varsNeg0.filterNot(x => insts.get(x.x).contains(Univ))
-
-      // Add new instantiations.
-      var currentInsts = insts ++
-        setElemOneOpt(elemPos, Empty) ++
-        setElemOneOpt(elemNeg, Univ) ++
-        cstsPos.iterator.map(_.c -> Empty) ++
-        cstsNeg.iterator.map(_.c -> Univ) ++
-        varsPos.iterator.map(_.x -> Empty) ++
-        varsNeg.iterator.map(_.x -> Univ)
-
-      // Recursively instantiate `other` while collecting further instantiations as we go.
-      val other = mutable.ListBuffer.empty[SetFormula]
-      for (f <- other0) {
-        val fProp = propagationWithInsts(f, currentInsts)
-        // Add elements, constants, and variables to the instantiations.
-        fProp match {
-          case Cst(c) =>
-            currentInsts += (c -> Empty)
-            other.append(fProp)
-          case Compl(Cst(c)) =>
-            currentInsts += (c -> Univ)
-            other.append(fProp)
-          case Var(x) =>
-            currentInsts += (x -> Empty)
-            other.append(fProp)
-          case Compl(Var(x)) =>
-            currentInsts += (x -> Univ)
-            other.append(fProp)
-          case e@ElemSet(_) =>
-            currentInsts ++= setElemOne(e, Empty)
-            other.append(fProp)
-          case Compl(e@ElemSet(_)) =>
-            currentInsts ++= setElemOne(e, Univ)
-            other.append(fProp)
-          case Univ => return Univ
-          case Empty => () // Redundant, so don't include.
-          case Compl(_) => other.append(fProp)
-          case Inter(_, _, _, _, _, _, _) => other.append(fProp)
-          case Union(_, _, _, _, _, _, _) => other.append(fProp)
-          case Xor(_) => other.append(fProp)
-        }
-      }
-      mkUnionAll(subformulasOf(elemPos, cstsPos, varsPos, elemNeg, cstsNeg, varsNeg, other).toList)
+    case Union(l) => Union(l)
```

Comment:
This is like chopping one leg off of propagation, I think it should either be removed or be implemented for union.

Reply:
I am planning to remove propagation. So this is just temporary.